### PR TITLE
feat: Story 1.2.1 - RubyLLM Integration with Parameter Management

### PR DESCRIPTION
**Pull request by sevos**
## Summary
Implements Story 1.2.1: RubyLLM Integration, providing a comprehensive LLM provider layer for the Catalyst framework with robust parameter management and execution tracking.

## Key Features Implemented
- **RubyLLM Integration**: Added ruby_llm (~> 1.3) as dependency with Rails 8 compatibility
- **Multi-Provider Support**: Comprehensive initializer supporting OpenAI, Anthropic, and Gemini providers  
- **Execution Enhancement**: Added chat-like fields (interaction_count, last_interaction_at, input_params)
- **Reusable Concerns**: Extracted InputParameterizable concern for parameter management across models
- **Generator Updates**: Enhanced install generator with RubyLLM configuration and clear setup instructions

## Acceptance Criteria Met ✅
1. ✅ RubyLLM gem added as dependency to Catalyst engine
2. ✅ Install generator creates comprehensive RubyLLM configuration initializer  
3. ✅ Initializer reads API keys from environment variables with Rails credentials fallback
4. ✅ Execution model has chat-like fields for tracking interactions without storing messages
5. ✅ Default configuration uses OpenAI provider with gpt-4.1-nano model
6. ✅ Execution model stores input parameters with JSON serialization similar to Agent model_params
7. ✅ Generator output clearly indicates next steps for API key configuration

## Technical Implementation
- **Database Schema**: Added 3 new fields to catalyst_executions table with proper migrations
- **Model Enhancement**: Execution model includes InputParameterizable concern with 6 helper methods
- **Configuration Management**: Environment variables with Rails credentials fallback under :catalyst namespace
- **SQLite Compatibility**: Text fields with JSON serialization for cross-database support
- **Security**: Proper API key handling without hardcoded secrets

## Test Coverage
- **90 tests pass** with 395 assertions, 0 failures
- **Comprehensive coverage** including concern isolation testing
- **Generator tests** verify proper file creation and configuration
- **Model tests** cover all new functionality including edge cases
- **Integration tests** ensure compatibility with existing features

## Test plan
- [x] All existing tests continue to pass
- [x] New RubyLLM integration tests added and passing
- [x] Generator creates proper initializer files
- [x] Execution model parameter management works correctly  
- [x] InputParameterizable concern functions independently
- [x] Migration templates include new fields
- [x] Environment variable configuration works as expected
- [x] Rails credentials fallback operates correctly

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: app/models/concerns/catalyst/input_parameterizable.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    # Helper methods for input parameter management
+    def input_parameters
+      input_params || {}
+    end
+
+    def input_parameters=(params)
+      self.input_params = params&.transform_keys(&:to_s)
+    end
+
+    # Helper method to get specific input parameter
```
Comment (sevos) [inline app/models/concerns/catalyst/input_parameterizable.rb:19]: I am not big fan of these helper methods, maybe we don't need a concern at all. Input params is just to store the input parameter values as json in the table. Remove the following helpers.

Additionally, consider removing this concern, if it would be composed only of a single `serialize` call.

Additionally, repeat above considerations and relevant comments for ModelParameterizable concern 

:point_up: each refactor in separate commit (TDD approach)
